cellpack/autopack/AWSHandler.py

```diff
@@ -89,10 +89,11 @@ def create_presigned_url(self, object_name, expiration=3600):
         # https://{self.bucket_name}.s3.{region}.amazonaws.com/{object_key}
         return url
 
-    def save_file(self, file_path):
+    def save_file_and_get_url(self, file_path):
         """
         Uploads a file to S3 and returns the presigned url
         """
         file_name = self.upload_file(file_path)
         if file_name:
-            return file_name, self.create_presigned_url(file_name)
+            base_url = self.create_presigned_url(file_name).split("?")[0]
```

Comment:
Nice!

Reply:
I might push this into the the `create_presigned_url` function so the function returns a url that the next function knows it can use without any adjustments. I also think this is a good bug for some tests. Ie make a list of urls that you know would be a problem and make sure we catch all of them.